### ISSUE TITLE
MlpPolicy vs CnnPolicy for Custom Architecture

### ISSUE THREAD

**Issue by anilkurkcu** [DLR-RM/stable-baselines3#1141]
### ❓ Question

Hello,

I was wondering what is the difference between setting the model as 'MlpPolicy' vs. setting it with 'CnnPolicy' when using a custom neural network architecture. My custom network is a 1-D convolutional neural network, and specifying the model as Mlp or Cnn both work without giving any errors.

So is there any difference between these two as long as we use our own custom neural network architecture?

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by qgallouedec** [comment]
By "custom neural network architecture" you mean custom feature extractor right?

**Comment by anilkurkcu** [comment]
Well yes, but I am not sure why that would differ from a custom NN architecture.

**Comment by qgallouedec** [comment]
Then the answer is no:

```python
PPO("MlpPolicy", env, policy_kwargs=dict(features_extractor_class=MyFeaturesExtractor))
```

and

```python
PPO("CnnPolicy", env, policy_kwargs=dict(features_extractor_class=MyFeaturesExtractor))
```

are equivalent.

(PPO for the example, but true for every algo)

TLDR:
CnnPolicy is just used to overwrite the default features extractor class of MlpPolicy. If you set your custom features extractor, then both are equivalent.

**Comment by araffin** [comment]
To add a bit more context, this is how `CnnPolicy` is implemented:

https://github.com/DLR-RM/stable-baselines3/blob/d5d1a02c15cdce868c72bbc94913e66fdd2efd3a/stable_baselines3/sac/policies.py#L396

it just calls the same constructor as `MlpPolicy` but with a different default feature extractor.
